easybuild/framework/easyblock.py

```diff
@@ -4198,6 +4207,98 @@ def print_dry_run_note(loc, silent=True):
     dry_run_msg(msg, silent=silent)
 
 
+def persist_failed_compilation_log_and_artifacts(build_successful, application_log, silent, app, easyconfig):
+    persistent_dirs = {}
+
+    def get_dir(path, dirname):
+        nonlocal persistent_dirs
+        dirpath = persistent_dirs.get(dirname)
+        if dirpath is None:
+            dirpath = create_unused_path(path)
+            persistent_dirs[dirname] = dirpath
+        return persistent_dirs[dirname]
+
+    def get_artifact_dir(path):
+        return get_dir(path, 'artifact')
+
+    def get_log_dir(path):
+        return get_dir(path, 'log')
+
+    def initialize_persistent_dir_and_replicate_if_paths_distinct(replication_operation,
+                                                                  get_directory,
+                                                                  source_path,
+                                                                  destination_base_path,
+                                                                  destination_relative_path):
+        abs_source_path = os.path.abspath(source_path)
+        abs_destination_base_path = os.path.abspath(destination_base_path)
+        abs_destination_path = os.path.join(abs_destination_base_path, destination_relative_path)
+
+        if not os.path.exists(abs_source_path):
+            return
+        if os.path.realpath(abs_source_path) == os.path.realpath(abs_destination_path):
+            return
+
+        final_abs_destination_base_path = get_directory(destination_base_path)
+        final_abs_destination_path = os.path.join(final_abs_destination_base_path, destination_relative_path)
+
+        replication_operation(abs_source_path, final_abs_destination_path)
+
+    if not application_log:
+        return
+
+    # there may be multiple log files, or the file name may be different due to zipping
+    logs = glob.glob('%s*' % application_log)
+    print_msg(
+        "Results of the build can be found in the temporary log file(s) %s" % ', '.join(logs),
+        log=_log,
+        silent=silent
+    )
+
+    if build_successful:
+        return
+
+    datetime_stamp = time.strftime("%Y%m%d") + '-' + time.strftime("%H%M%S")
+
+    errorlog_path = get_log_error_path(ec=easyconfig)
+    if errorlog_path is not None:
+        errorlog_path = os.path.join(errorlog_path, datetime_stamp)
+
+        for log_file in logs:
+            initialize_persistent_dir_and_replicate_if_paths_distinct(
+                copy_file,
+                get_log_dir,
+                log_file,
+                errorlog_path,
+                os.path.basename(log_file)
```

Comment:
The "if_paths_distinct" here is:
`log_file == errorlog_path / basename(log_file)` or `dirname(application_log) == errorlog_path`

In which situation can this be true? And especially as `errorlog_path` includes the timestamp with seconds I'd argue it is never the same